### PR TITLE
Start CI migration to Jenkins pipelines

### PR DESCRIPTION
**Pull request by ArsenArsen**
Possible defect: the two rsync steps towards the end could make the builds fail on computers of developers not set up right. This is a problem only for those who wish to maintain CI

### CODE REVIEW

File: ci/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,48 @@
+node {
+    cleanWs()
+    dir('src') { git 'https://github.com/managarm/bootstrap-managarm.git' }
+    docker.build('managarm_cienv', 'src/ci/docker/').inside {
+        dir('build') {
+            stage('Build system') {
+                sh '''#!/bin/sh
+                set -xe
+                xbstrap init ../src || true
```
Comment (ArsenArsen) [inline ci/Jenkinsfile]: quick remark about this line: `|| true` is unnecessary if `cleanWs()` is used. I used it to speed up prototyping. It's not mandatory, but it should have little to no effect as the only effect `init` has is a symlink

File: ci/Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,51 @@
+node {
+    cleanWs()
+    dir('src') { checkout scm }
+    def userId = sh(returnStdout: true, script: 'id -u').trim()
+    echo "${userId}"
```
Comment (avdgrinten) [inline ci/Jenkinsfile]: Left over from debugging?

File: docker/Dockerfile [hunk 3]
```diff
@@ -1,16 +1,18 @@
...
 
-RUN useradd -ms /bin/bash managarm_buildenv
+RUN useradd -ms /bin/bash managarm_buildenv -u $USER
 USER managarm_buildenv
 WORKDIR /home/managarm_buildenv
 
 RUN git config --global user.email "managarm_buildenv@localhost"
 RUN git config --global user.name "managarm_buildenv"
 
+ADD gitconfig /etc/gitconfig
```
Comment (avdgrinten) [inline docker/Dockerfile]: Why do we still need this?

File: ci/Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+                mkdir -p hel
+                sed 's|@ROOTDIR@|../src/managarm/hel|' ../src/managarm/hel/Doxyfile.in > Doxyfile
+                doxygen
+                '''
+            }
+        }
+    }
+
+    stage('Make image') {
+        withEnv(['LIBGUESTFS_BACKEND_SETTINGS=force_tcg']) {
```
Comment (avdgrinten) [inline ci/Jenkinsfile]: We don't need `force_tcg` anymore.

File: docker/Dockerfile (hunk 3 above)
Comment (ArsenArsen) [reply docker/Dockerfile]: It can probably be removed without losing anything

**Pull request merged**